src/runtime.jl

```diff
@@ -2,13 +2,13 @@ struct RuntimeDevice{D}
     device::D
 end
 default_device() = RuntimeDevice(default_device(RUNTIME[]))
-default_device(::typeof(HSA)) = HSARuntime.get_default_agent()
+default_device(::DeviceRuntime) = HSARuntime.get_default_agent()
```

Comment:
This sort of stuff should be left as `typeof(HSA_rt)`, because when I eventually get OpenCL support working, we wouldn't be using HSARuntime.

Reply:
I don't mind reverting, but I'm a little confused. Since `HSA_rt`and `OCL_rt` are both enumerations of `DeviceRuntime` doesn't that make them both evaluate to the same thing when you call `typeof()` ?? Or is this a nuance of dispatch I'm not aware of?

Reply:
I've reverted this since it makes no difference for the time being.